friendshipper/src-tauri/src/repo/operations/gh/submit.rs

```diff
@@ -89,16 +89,38 @@ impl Task for GitHubSubmitOp {
             .send()
             .await?;
 
-        pr = self
-            .poll_for_mergeable(octocrab, pr, owner.clone(), repo.clone())
-            .await?;
-
-        let id = self
-            .client
-            .get_pull_request_id(owner.clone(), repo.clone(), pr.number as i64)
-            .await?;
-
-        self.client.enqueue_pull_request(id).await?;
+        let octocrab_clone = octocrab.clone();
+        let owner_clone = owner.clone();
+        let repo_clone = repo.clone();
+        let client_clone = self.client.clone();
+        let self_clone = self.clone();
+
+        // There's a lot of variability in how long this takes, so we give the frontend
+        // a chance to return control to the user before it starts polling
+        tokio::spawn(async move {
+            match self_clone
+                .poll_for_mergeable(octocrab_clone, pr, owner_clone.clone(), repo_clone.clone())
+                .await
+            {
+                Ok(updated_pr) => {
+                    if let Ok(id) = client_clone
+                        .get_pull_request_id(
+                            owner_clone.clone(),
+                            repo_clone.clone(),
+                            updated_pr.number as i64,
+                        )
+                        .await
+                    {
+                        if let Err(e) = client_clone.enqueue_pull_request(id).await {
+                            warn!("Failed to enqueue pull request: {:?}", e);
+                        }
+                    } else {
+                        warn!("Failed to get pull request ID");
+                    }
+                }
+                Err(e) => warn!("Failed to poll for mergeable state: {:?}", e),
+            }
+        });
```

Comment:
The potential downside here is if queuing the PR errors, the frontend might not be aware of it. I'd like any reviewer to chime in on whether that risk is worth the 10-30s of savings on submission time.

Reply:
I dont think the saved time is enough to offset a bad ui experience?
maybe if users were used to checking on their submissions via github, this would be fine.

Reply:
the bad ui experience is if and only if putting the PR into the queue errors out. the frontend still watches the PR and updates its state accordingly. the difference is that we were blocking the submission modal from closing until the PR gets queued. 

If the background thread errors out, users will see that their PR is open but not in the queue. The fix in this case would be to click the PR link in the UI and manually put the PR into the queue. In our data, this error has occurred 0 times in the last 60 days.

Reply:
this seems fine then